pandas/_config/config.py

```diff
@@ -820,6 +821,34 @@ def inner(x):
     return inner
 
 
+def is_pos_int():
+    """
+    Creates a function for validating that an option value is either a positive
+    integer or a None type
+
+    Returns
+    -------
+    validator - a function of a single argument x , which raises
+                ValueError if type(x) is not equal to `_type` or if x is not
+                positive in the case of a int type.
+
+    """
+
+    _type = (type(None), int)
+    type_repr = "|".join(map(str, _type))
+
+    def inner(x):
+        if not isinstance(x, _type):
```

Comment:
I think you could simplify this if you just used a condition like `if not (x is None or isinstance(x, int))`

Reply:
Agreed,  since the simplified function isn't really dynamic, the latest commit modified is to just be called standalone.